src/app/payment/views/IntegratedCheckoutView/CheckoutViewWrapper.tsx

```diff
@@ -0,0 +1,221 @@
+import { useEffect, useCallback, useReducer, useState } from 'react';
+import { Elements } from '@stripe/react-stripe-js';
+
+import navigationService from '../../../core/services/navigation.service';
+import { AppView } from '../../../core/types';
+import errorService from '../../../core/services/error.service';
+import CheckoutView from './CheckoutView';
+import envService from '../../../core/services/env.service';
+import { StripeElementsOptions, loadStripe } from '@stripe/stripe-js';
+import databaseService from '../../../database/services/database.service';
+import localStorageService from '../../../core/services/local-storage.service';
+import RealtimeService from '../../../core/services/socket.service';
+import authCheckoutService from '../../services/auth-checkout.service';
+import { useAppDispatch } from '../../../store/hooks';
+import { useSignUp } from '../../../auth/components/SignUp/useSignUp';
+import { AuthMethodTypes, ErrorType, THEME_STYLES } from '../../types';
+import { checkoutReducer, initialStateForCheckout } from './checkoutReducer';
+import checkoutService from 'app/payment/services/checkout.service';
+import { useThemeContext } from 'app/theme/ThemeProvider';
+
+export const stripePromise = (async () => {
+  const stripeKey = envService.isProduction() ? process.env.REACT_APP_STRIPE_PK : process.env.REACT_APP_STRIPE_TEST_PK;
+  return await loadStripe(stripeKey);
+})();
+
+const CheckoutViewWrapper = () => {
+  const { checkoutTheme } = useThemeContext();
+  const dispatch = useAppDispatch();
+  const { doRegister } = useSignUp('activate');
+
+  const [state, dispatchReducer] = useReducer(checkoutReducer, initialStateForCheckout);
+  const [isUpsellSwitchActivated, setIsUpsellSwitchActivated] = useState<boolean>(false);
+
+  const { authMethod, error, currentPlanSelected, plan, stripe, couponCodeData, elementsOptions, promoCodeName } =
+    state;
+  const { backgroundColor, textColor } = THEME_STYLES[checkoutTheme as string];
+
+  useEffect(() => {
+    const params = new URLSearchParams(window.location.search);
+    const planId = params.get('planId');
+    const promotionCode = params.get('promotion_code');
+
+    const elementsOptionsParams: StripeElementsOptions = {
```

Comment:
I think there is no need for `elementsOptionsParams` to be inside the `useEffect`